crates/transaction/models/src/low_level/manifest_encountered_component_address.rs

```diff
@@ -10,20 +10,19 @@ address_union!(
 
 macro_rules! impl_try_from_for_manifest_encountered_address {
     ($($variant:ident => $address_type:ty),*) => {
-        impl TryFrom<(ScryptoGlobalAddress, NetworkID)> for ManifestEncounteredComponentAddress {
+        impl TryFrom<(ScryptoManifestAddress, NetworkID)> for ManifestEncounteredComponentAddress {
             type Error = CommonError;
 
-            fn try_from(value: (ScryptoGlobalAddress, NetworkID)) -> Result<Self> {
-                let (global_address, network_id) = value;
+            fn try_from(value: (ScryptoManifestAddress, NetworkID)) -> Result<Self, Self::Error> {
+                let (address, network_id) = value;
 
                 $(
-                    if let Ok(address) = <$address_type>::try_from((global_address, network_id)) {
+                    if let Ok(address) = <$address_type>::try_from(value) {
                         return Ok(ManifestEncounteredComponentAddress::$variant(address));
                     }
                 )*
-
                 Err(CommonError::FailedToCreateAddressFromGlobalAddressAndNetworkID {
```

Comment:
Maybe rename the error to `FailedToCreateAddressFromManifestAddressAndNetworkID { address: address.to_string() }` or `to_hex`